### PR TITLE
Added types for olefile

### PR DESCRIPTION
**Pull request by radarhere**
Adding types for https://pypi.org/project/olefile

### CODE REVIEW

**Review by srittau — COMMENTED** [review]
Thanks! Not a full review, but I noticed you set `partial_stub` and `ignore_missing_stub` to `true`, although as far as I can see the package only contains the two files for which you included stubs. Any particular reason for that?

I also think that you can remove it from the excludes in `pyright.stricter.json` as you have fully annotated the stubs!

**Comment by radarhere** [comment]
Thanks. I was able to remove it from excludes in `pyright.stricter.json`, but when I try to say this isn't a partial stub, I get https://github.com/python/typeshed/actions/runs/7915896331/job/21608670958?pr=11416
> error: olefile.olefile2 failed to find stubs

I can't find `olefile2` in the source code, so I don't actually know why that is.

**Comment by srittau** [comment]
That's weird indeed. Their seem to be some artifacts in the released package. But a better way to work around this problems is to add an `olefile\.olefile2.*` entry to a new file `@tests/stubtest_allowlist.txt`.

**Comment by radarhere** [comment]
Thanks, done.

File: stubs/olefile/olefile/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import datetime
```
Comment (JelleZijlstra) [inline stubs/olefile/olefile/__init__.pyi:1]: At runtime this file just has `from .olefile import *`. The stub should do the same. Currently, it instead duplicates all the definitions from `olefile.olefile`.

Comment (radarhere) [reply stubs/olefile/olefile/__init__.pyi]: Thanks, I've pushed a commit for this.

File: stubs/olefile/olefile/olefile.pyi [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    DIRENTRY_SIZE: int
+
+    def __init__(self, entry: bytes, sid: int, ole_file: OleFileIO) -> None: ...
+    def build_sect_chain(self, ole_file: OleFileIO) -> None: ...
+    def build_storage_tree(self) -> None: ...
+    def append_kids(self, child_sid: int) -> None: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    def __le__(self, other: object) -> bool: ...
```
[the remark this answers is not in the document]
Comment (radarhere) [reply stubs/olefile/olefile/olefile.pyi:169]: If I attempt this, mypy complains - https://github.com/radarhere/typeshed/actions/runs/7939554536/job/21679821620#step:5:96
> error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stubs/olefile/olefile/olefile.pyi:169]: Then you need to type ignore that. The runtime has an incompatible override, so we need to reflect that in the stub too.

[the remark this answers is not in the document]
Comment (radarhere) [reply stubs/olefile/olefile/olefile.pyi]: Ok, I've pushed that as a commit.

File: stubs/olefile/olefile/olefile.pyi [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    ) -> None: ...
+    def _decode_utf16_str(self, utf16_str: bytes, errors: str = "replace") -> bytes: ...
+    def open(self, filename: IO[bytes] | bytes | str, write_mode: bool = False) -> None: ...
+    def close(self) -> None: ...
+    def _close(self, warn: bool = False) -> None: ...
+    def _check_duplicate_stream(self, first_sect: int, minifat: bool = False) -> None: ...
+    def dumpfat(self, fat: Sequence[int], firstindex: int = 0) -> None: ...
+    def dumpsect(self, sector: bytes, firstindex: int = 0) -> None: ...
+    def sect2array(self, sect: bytes) -> Sequence[int]: ...
+    def loadfat_sect(self, sect: bytes | list[int]) -> int | None: ...
```
[the remark this answers is not in the document]
Comment (radarhere) [reply stubs/olefile/olefile/olefile.pyi]: Ok, I've pushed a commit for this.

**Review by JelleZijlstra — APPROVED** [review]
The current CI failure is unrelated, I will merge after that's resolved.

**Pull request merged**